src/platform/nm-linux-platform.c

```diff
@@ -5935,16 +5950,25 @@ link_set_sriov_params (NMPlatform *platform,
 		return FALSE;
 	}
 
-	if (!nm_platform_sysctl_set (NM_PLATFORM_GET,
-	                             NMP_SYSCTL_PATHID_NETDIR (dirfd,
-	                                                       ifname,
-	                                                       "device/sriov_numvfs"),
-	                             nm_sprintf_buf (buf, "%u", num_vfs))) {
-		_LOGW ("link: couldn't set SR-IOV num_vfs to %d: %s", num_vfs, strerror (errno));
-		return FALSE;
+	if (current_num == 0 && num_vfs == 0) {
+		if (callback) {
+			packed = nm_utils_user_data_pack (g_object_ref (platform),
+			                                  callback,
+			                                  data);
+			nm_utils_invoke_on_idle (sriov_idle_cb, packed, cancellable);
+		}
```

Comment:
`nm_platform_sysctl_set_async()` returns a boolean to indicate whether the action was successfully started. On the one hand, I dislike this inconsistency (an async operation should never fail to start). On the other hand, it's actually simpler to implement and avoids an unnecessary idle action, so OK.

But here, you have  a case where you do invoke a trivial response in an idle handler. Either always schedule async actions (for consistency), or don't schedule such idle actions (for efficiency).

Reply:
Fixed, the functions now always return asynchronously.